### PR TITLE
feat: added OutputTable, removed externalIds and random bugs

### PR DESCRIPTION
**Pull request by calebjclark**
- hero.close now returns a promise, even if it's already in the process of closing. Same with ConnectionManager#close.
- removed the concept of externalIds
- Added OutputTable to SessionDb.
- Fixed some issues with Fullstack. The big one being that Core should NOT be started just because the file is imported.

### CODE REVIEW

File: client/lib/ConnectionManager.ts [hunk 1]
```diff
@@ -4,17 +4,19 @@ import ConnectionToCore from '../connections/ConnectionToCore';
 import CoreSession from './CoreSession';
 import ConnectionFactory from '../connections/ConnectionFactory';
 import Hero, { IState, IStateOptions } from './Hero';
+import DisconnectedFromCoreError from '../connections/DisconnectedFromCoreError';
```
Comment (blakebyrnes) [inline client/lib/ConnectionManager.ts:7]: lint issue

File: core/connections/ConnectionToClient.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import Core from '../index';
 import FrameEnvironment from '../lib/FrameEnvironment';
 import CommandRunner, { ICommandableTarget } from '../lib/CommandRunner';
 import RemoteEvents from '../lib/RemoteEvents';
+import { IOutputChangeRecord } from '../models/OutputTable';
```
Comment (blakebyrnes) [inline core/connections/ConnectionToClient.ts:19]: lint

File: fullstack/package.json [hunk 3]
```diff
@@ -11,7 +11,8 @@
     "@ulixee/hero": "1.5.4",
     "@ulixee/hero-core": "1.5.4",
     "@ulixee/hero-interfaces": "1.5.4",
-    "@ulixee/execute-js-plugin": "1.5.4"
+    "@ulixee/execute-js-plugin": "1.5.4",
+    "awaited-dom": "1.3.1"
```
Comment (blakebyrnes) [inline fullstack/package.json:15]: Do we want this to be a direct dependency? If we're only pulling types, you can leave it as a devDependency

**Comment by blakebyrnes** [comment]
This looks good. Just have a few lint issues and a question about awaited-dom as a dependency.

File: fullstack/lib/startCore.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import Core, { GlobalPool } from '@ulixee/hero-core';
+import ShutdownHandler from '@ulixee/commons/lib/ShutdownHandler';
+
+let hasStarted = false;
+
+export default function startCore(): void {
+  if (hasStarted) return;
```
Comment (blakebyrnes) [inline fullstack/lib/startCore.ts:7]: StartCore doesn't seem to actually start core...

File: fullstack/package.json (hunk 3 above)
Comment (calebjclark) [reply fullstack/package.json:15]: I was getting squiggly lines in Webstorm telling me I needed to make it a dependency. I can move it back.

File: fullstack/lib/startCore.ts (hunk 4 above)
Comment (calebjclark) [reply fullstack/lib/startCore.ts:7]: Ha. You're right. I was moving way too fast on that.

Importing fullstack into a test was stopping Jest from closing until I moved this code. I have no idea why.

File: client/lib/ConnectionManager.ts (hunk 1 above)
Comment (calebjclark) [reply client/lib/ConnectionManager.ts]: fixed

File: core/connections/ConnectionToClient.ts (hunk 2 above)
Comment (calebjclark) [reply core/connections/ConnectionToClient.ts]: fixed

File: fullstack/package.json (hunk 3 above)
Comment (calebjclark) [reply fullstack/package.json]: Fixed

File: fullstack/lib/startCore.ts (hunk 4 above)
Comment (calebjclark) [reply fullstack/lib/startCore.ts]: Reverted back to original.

**Pull request merged**